src/Parser/BalanceDruid/Modules/Features/AstralPower.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import Icon from 'common/Icon';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Module from 'Parser/Core/Module';
+import SPELLS from 'common/SPELLS';
+
+class AstralPower extends Module {
+  
+  lastAstral = 0;
+  aspWasted = 0;
+
+  on_toPlayer_energize(event) {
+      for (let i = 0; i < event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
+              const maxAsP = event.classResources[i].max / 10;
+              const addedAsP = event.resourceChange;
+
+              if (this.lastAstral + addedAsP > maxAsP){
+                  this.aspWasted += this.lastAstral + addedAsP - maxAsP;
+              }
+
+              this.lastAstral = event.classResources[i].amount / 10;
+          }
+      }
+  }
+
+  on_byPlayer_cast(event) {
+      const spellId = event.ability.guid;
+      if (SPELLS.STARSURGE_MOONKIN.id !== spellId && SPELLS.STARFALL.id !== spellId && SPELLS.STARFALL_CAST.id !== spellId) {
+          return;
+      }
+      
+      for (let i = 0; i<event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
+              if (event.classResources[i].cost) {
+                  this.lastAstral = this.lastAstral - (event.classResources[i].cost / 10);
+              }
+          }
+      }
+  }
+
+  suggestions(when) {
+      when(this.aspWasted).isGreaterThan(0)
+        .addSuggestion((suggest, actual, recommended) => {
+          return suggest(<span>You overcapped {this.aspWasted} Astral Power.</span>)
```

Comment:
Please see the document I just linked for recommended suggestion formatting. I'd like to suggest writing this something like `You wasted Astral Power. Try to use all your Astral Power, the ability doesn't matter so long as you don't waste any.`. (I'd prefer recommending a specific spell as filler, but this is in line with what you wrote and I don't know how it works)